pandas/io/_util.py

```diff
@@ -21,3 +23,9 @@ def _arrow_dtype_mapping() -> dict:
         pa.float32(): pd.Float32Dtype(),
         pa.float64(): pd.Float64Dtype(),
     }
+
+
+def arrow_string_types_mapper() -> Callable:
+    pa = import_optional_dependency("pyarrow")
+
+    return {pa.string(): pd.ArrowDtype(pa.string())}.get
```

Comment:
Thinking about this a little, is there a situation where you would want to mix pyarrow and numpy dtypes?

(I'm thinking maybe we should force users to pick the pyarrow dtype backend if you are using the pyarrow string type)

Reply:
Yes there are a lot of situations.

NumPy numeric and Arrow strings is still the fastest, numpy numeric is 2D. Forcing them right now is not a good idea